subsys/bluetooth/host/gatt.c

```diff
@@ -711,9 +704,13 @@ static ssize_t db_hash_read(struct bt_conn *conn,
 	/* Check if db_hash is already pending in which case it shall be
 	 * generated immediately instead of waiting for the work to complete.
 	 */
-	err = k_delayed_work_cancel(&db_hash_work);
-	if (!err) {
-		db_hash_gen(true);
+	err = k_work_delayable_busy_get(&db_hash_work);
```

Comment:
I don't particularly like `err` here, since the busy flag state can never indicate an error, and will never be negative (indicating an error).

    if (k_work_delayable_busy_get(&db_hash_work) != 0) {

would be a little less misleading.  Or just declare a new `int` variable `res` or `busy` or whatever; it'll probably never move out of its register.

Same comment applies around line 5000.

Reply:
Use of err is removed